lib/sea.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const { isSea, getAsset: getAssetInternal } = internalBinding('sea');
+const { TextDecoder } = require('internal/encoding');
+const { validateString } = require('internal/validators');
+const {
+  ERR_NOT_IN_SINGLE_EXECUTABLE_APPLICATION,
+  ERR_SINGLE_EXECUTABLE_APPLICATION_ASSET_NOT_FOUND,
+} = require('internal/errors').codes;
+const { Blob } = require('internal/blob');
+
+function getRawAsset(key) {
+  validateString(key, 'key');
+
+  if (!isSea()) {
+    throw new ERR_NOT_IN_SINGLE_EXECUTABLE_APPLICATION();
```

Comment:
Can we throw this error from C++?

Reply:
Unless this is ends up being performance-senstive (I'd assume for SEA this would rarely be called because there should be a high-level thing e.g. VFS built on top on a single asset for that purpose), I'd prefer to keep it in JS for now since we have to go into JS for TextEncoder/Blob anyway.